### ISSUE TITLE
implement `Clone` for iterator types

### ISSUE THREAD

**Issue by mikeyhew** [bodil/im-rs#134]
`std::vec::Iter<'a, T>` implements `Clone`, but it seems that `im::vector::Iter<'a, T>` does not. Would it be possible to to implement `Clone` for all of the by-reference `Iter` types in `im`, and would you accept a PR adding the impls?

**Comment by bodil** [comment]
Absolutely. There's no good reason why they shouldn't implement `Clone`.

**Comment by vorner** [comment]
I've hit the need of them just today. @mikeyhew are you working on the PR, or should I try to find a bit of time to give them a try?

**Comment by mikeyhew** [comment]
@vorner I'm not working on a PR for this, all yours!
